### ISSUE TITLE
Support `@ReadOnlyProperty`.

### ISSUE THREAD

**Issue by michael-simons** [spring-projects/spring-data-neo4j#2294]
We already have a similar machinery in place for `saveAs` as projection, so that is - in contrast to SDN5+OGM - much easier to implement. A couple of scenarios would benefit a lot from it. 

The question was brought up here https://github.com/spring-projects/spring-data-neo4j/issues/2289 for the first time.

**Comment by michael-simons** [comment]
Paging you here as well @BenBohannon. This is implemented in 6.1.2 and upwards. For your domain, this would be the solution I prefer. 

Snapshots at

```xml
<repositories>
  <repository>
    <id>spring-snapshots</id>
    <name>Spring Snapshots</name>
    <url>https://repo.spring.io/snapshot</url>
    <releases>
      <enabled>false</enabled>
    </releases>
  </repository>
</repositories>
```

**Comment by BenBohannon** [comment]
I can't get 6.1.2-SNAPSHOT or 6.2.0-SNAPSHOT to correctly load the repository beans to test them.

6.1.2 complains about the Repositories not being able to find `Mono` or `Flux` classes. Even installing the reactor dependency just leads to another error.

6.2.0 complains about `Caused by: java.lang.NoSuchMethodError: 'java.lang.Object org.springframework.data.neo4j.repository.support.Neo4jRepositoryFactory.instantiateClass(java.lang.Class, java.lang.Object[])'`

Temporarily, I was able to use saveAs to save a projection that hide the incoming relations and only saves outgoing relations. But that only works when saving a single Node. Saving a different type of node seems to cascade the save to the node that has a relation to the node with incoming and outgoing relations, and that doesn't use the projection to save.
